translation-service-eTranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java

```diff
@@ -0,0 +1,211 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.springframework.data.redis.listener.ChannelTopic;
+import org.springframework.data.redis.listener.RedisMessageListenerContainer;
+import org.springframework.data.redis.listener.adapter.MessageListenerAdapter;
+import eu.europeana.api.translation.definitions.model.TranslationObj;
+import eu.europeana.api.translation.service.AbstractTranslationService;
+import eu.europeana.api.translation.service.exception.TranslationException;
+
+public class ETranslationTranslationService extends AbstractTranslationService {
+  
+  private static final Logger LOGGER = LogManager.getLogger(ETranslationTranslationService.class);
+
+  private String serviceId;
+  private final String baseUrl;
+  private final String domain;
+  //this is the base url of the translation api (without the request handler (or the controller) endpoint)
+  private final String callbackUrl;
+  private final String credentialUsername;
+  private final String credentialPwd;
+  private final int maxWaitMillisec;
+  private final RedisMessageListenerContainer redisMessageListenerContainer;
+  
+  public ETranslationTranslationService(String baseUrl, String domain, String callbackUrl, int maxWaitMillisec, 
+      String username, String password, RedisMessageListenerContainer redisMessageListenerContainer) {
+    this.baseUrl = baseUrl;
+    this.domain = domain;
+    this.callbackUrl=callbackUrl;
+    this.maxWaitMillisec=maxWaitMillisec;
+    this.credentialUsername=username;
+    this.credentialPwd=password;
+    this.redisMessageListenerContainer=redisMessageListenerContainer;
+  }
+
+  @Override
+  public void translate(List<TranslationObj> translationObjs) throws TranslationException {
+    //base64 encoded string, which will not be translated from eTransaltion, used as a markup delimiter
+    String markupDelimit=" ZXd3ZHdld2U= ";
+    String eTranslJointStr = generateJointStringForTranslation(translationObjs, markupDelimit);
+    
+    /* create an eTransl request with an external reference and send it. The same external reference is received
+     * in the eTransl callback. That reference is used for the name of the channel for the redis message subscriber 
+     * listener created below, which will be notified from the redis publisher after the eTransl callback comes.
+     * The publisher will publish to the same channel using the external reference from the eTransl callback.
+     */
+    if(StringUtils.isNotBlank(eTranslJointStr)) {
```

Comment:
better check if the translationObjs is empty at the beginning, can also use the @nonnull annotation